### ISSUE TITLE
Synology StorageClass Parameters

### ISSUE THREAD

**Issue by codello** [democratic-csi/democratic-csi#175]
Hello, I just started trying out this project using the `synology-iscsi` driver and it works really well so far.

However I feel that the configuration is currently a little inconvenient in that you have to configure the LUN and Target options globally. I think these options would go really well in a `StorageClass`. So instead of having
```yaml
…
driver: synology-iscsi
iscsi:
  lunTemplate:
    type: BLUN
  targetTemplate:
    max_sessions: 0
```
you could instead do something like this:
```yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: synology-iscsi-blun
…
parameters:
  fsType: ext4
  lunType: BLUN
  targetMaxSessions: 0
```

This way you could easily create multiple storage classes for different LUN types without requiring multiple CSI instances. Similarly one could leverage the `secrets` of the `StorageClass` resource to implement CHAP authentication per `StorageClass` (or even per LUN by using the templating features of the `StorageClass` secrets).

I'd be interested in creating a PR for this feature. Would something like this have a chance of being merged?

**Comment by travisghansen** [comment]
Welcome! I wasn’t sure if anyone actually used the synology driver, glad to hear it’s working well!

I’ll have to ponder the request a bit as there are ramifications that aren’t overly apparent on the surface. My initial inclination is that any of the template options probably would not be an issue supplied at the storage class level but let me audit that a little more thoroughly mentally.

**Comment by travisghansen** [comment]
This has been released as discussed in `v1.7.0`
